xrpl4j-client/src/test/java/org/xrpl/xrpl4j/client/XrplClientTest.java

```diff
@@ -0,0 +1,1256 @@
+      .account(Address.of("rvYAfWj5gh67oV6fW32ZzP3Aw4Eubs59Ba"))
+      .fee(XrpCurrencyAmount.ofDrops(UnsignedLong.ONE))
+      .sequence(UnsignedInteger.ONE)
+      .amount(XrpCurrencyAmount.ofDrops(100))
+      .destination(Address.of("rvYAfWj5gh67oV6fW32ZzP3Aw4Eubs59Ba"))
+      .settleDelay(UnsignedInteger.ONE)
+      .publicKey("123")
+      .build();
+    assertThat(xrplClient.addSignature(paymentChannelCreate, "sign").transactionSignature().get()).isEqualTo("sign");
+    PaymentChannelClaim paymentChannelClaim = PaymentChannelClaim.builder()
+      .account(Address.of("rvYAfWj5gh67oV6fW32ZzP3Aw4Eubs59Ba"))
+      .fee(XrpCurrencyAmount.ofDrops(UnsignedLong.ONE))
+      .sequence(UnsignedInteger.ONE)
+      .channel(Hash256.of("0123456789012345678901234567890123456789012345678901234567891234"))
+      .build();
+    assertThat(xrplClient.addSignature(paymentChannelClaim, "sign").transactionSignature().get()).isEqualTo("sign");
+    PaymentChannelFund paymentChannelFund = PaymentChannelFund.builder()
+      .account(Address.of("rvYAfWj5gh67oV6fW32ZzP3Aw4Eubs59Ba"))
+      .fee(XrpCurrencyAmount.ofDrops(UnsignedLong.ONE))
+      .sequence(UnsignedInteger.ONE)
+      .channel(Hash256.of("0123456789012345678901234567890123456789012345678901234567891234"))
+      .amount(XrpCurrencyAmount.ofDrops(100L))
+      .build();
+    assertThat(xrplClient.addSignature(paymentChannelFund, "sign").transactionSignature().get()).isEqualTo("sign");
+    SetRegularKey setRegularKey = SetRegularKey.builder()
+      .account(Address.of("rf1BiGeXwwQoi8Z2ueFYTEXSwuJYfV2Jpn"))
+      .fee(XrpCurrencyAmount.ofDrops(12))
+      .sequence(UnsignedInteger.ONE)
+      .regularKey(Address.of("rAR8rR8sUkBoCZFawhkWzY4Y5YoyuznwD"))
+      .build();
+    assertThat(xrplClient.addSignature(setRegularKey, "sign").transactionSignature().get()).isEqualTo("sign");
+    SignerListSet signerListSet = SignerListSet.builder()
+      .account(Address.of("rf1BiGeXwwQoi8Z2ueFYTEXSwuJYfV2Jpn"))
+      .fee(XrpCurrencyAmount.ofDrops(12))
+      .sequence(UnsignedInteger.ONE)
+      .signerQuorum(UnsignedInteger.valueOf(3))
+      .addSignerEntries(
+        SignerEntryWrapper.of(
+          SignerEntry.builder()
+            .account(Address.of("rsA2LpzuawewSBQXkiju3YQTMzW13pAAdW"))
+            .signerWeight(UnsignedInteger.valueOf(2))
+            .build()
+        )
+      )
+      .build();
+    assertThat(xrplClient.addSignature(signerListSet, "sign").transactionSignature().get()).isEqualTo("sign");
+    TicketCreate ticketCreate = TicketCreate.builder()
+      .account(Address.of("rf1BiGeXwwQoi8Z2ueFYTEXSwuJYfV2Jpn"))
+      .fee(XrpCurrencyAmount.ofDrops(UnsignedLong.ONE))
+      .sequence(UnsignedInteger.ONE)
+      .ticketCount(UnsignedInteger.ONE)
+      .build();
+    assertThat(xrplClient.addSignature(ticketCreate, "sign").transactionSignature().get()).isEqualTo("sign");
+  }
+
+  @Test
+  public void submitWithSignedTx() {
```

Comment:
This test looks good, but we're still missing coverage for the other variant, with this signature:

```
public <T extends Transaction> SubmitResult<T> submit(
    final org.xrpl.xrpl4j.crypto.signing.SignedTransaction signedTransaction
  )
```